pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -336,6 +334,17 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{}, err
 	}
 
+	// Recover Pods with updated init containers in case of fails
```

Comment:
Can't remove this now that the retry mechanism is part of the initContainer?

Reply:
No, this is another type of the recovery. It protects from the incorrect `sha` sum entering, not from `http` host unavailability.

Reply:
Yes, sure, @ryanemerson . The scenario is:
- create cluster with or without dependencies and correct forming
- add or change (add hash or url) dependencies
- starting cluster rolling upgrade
- init container fails and first Pod become invalid
- change somewhere in dependencies to fix previous errors
- this code kill the Pod, otherwise StatefulSet controller never rolling update filed Pods

Reply:
Shouldn't this code only be executed if a update is needed then? Currently this code will execute causing the following scenario:

- Create cluster with incorrect hash/url
- init-container fails
- Pods are deleted
- init-container fails again

Reply:
> Shouldn't this code only be executed if a update is needed then? Currently this code will execute causing the following scenario

No, comparing init container in Pod and propagated to StatefulSet are protects from this scenario.

Reply:
Thanks for explaining. I understand now.